core/pkg/model/custom_cost_request.go

```diff
@@ -0,0 +1,20 @@
+package model
+
+import (
+	"time"
+
+	"github.com/opencost/opencost/core/pkg/opencost"
+)
+
+type CustomCostRequest struct {
+	TargetWindow *opencost.Window
+	Resolution   time.Duration
+}
+
+func (c *CustomCostRequest) GetTargetWindow() *opencost.Window {
```

Comment:
Why have getters if all fields are exported and you aren't copying them?

Reply:
for the RPC, I think they have to implement an interface. I have just added that interface definition

Reply:
Ah, this probably answers the questions I just asked on the other PR. Perhaps we continue the discussion there; I'm curious about this requirement